test/podman-tests.sh

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env bash
+
+# Do not set -e as we want to work on all results of `rpm -q dnf[5]`.
+set -xo pipefail
+
+if [[ $1 == '' ]]; then
+    echo -e "Usage: podman-tests.sh TEST_TYPE STREAM\nTEST_TYPE can be 'e2e' or 'system'\nSTREAM can be 'upstream' or 'downstream'"
+    exit 0
+fi
+
+# `rhel` macro exists on RHEL, CentOS Stream, and Fedora ELN
+# `centos` macro exists only on CentOS Stream
+CENTOS_VERSION=$(rpm --eval '%{?centos}')
+RHEL_VERSION=$(rpm --eval '%{?rhel}')
+
+# For upstream tests, we need to test with podman and other packages from the
+# podman-next copr. For downstream tests (bodhi, errata), we don't need any
+# additional setup
+if [[ $2 == "upstream" ]]; then
+    # Use CentOS Stream 10 copr target for RHEL-10 until EPEL 10 becomes
+    # available
+    if [[ -n $CENTOS_VERSION || $RHEL_VERSION -ge 10 ]]; then
+        dnf -y copr enable rhcontainerbot/podman-next centos-stream-$CENTOS_VERSION
+    else
+        dnf -y copr enable rhcontainerbot/podman-next
+    fi
+    echo "priority=5" >> /etc/yum.repos.d/_copr:copr.fedorainfracloud.org:rhcontainerbot:podman-next.repo
+fi
+
+# Remove testing-farm repos if they exist
+rm -f /etc/yum.repos.d/tag-repository.repo
```

Comment:
Can add more comments on why this one is needed?

Reply:
because it's a PITA? :)  . I've found the testing-farm repos to always interfere with the packages I want to use. Things are a lot easier if I just delete it.

Reply:
Lol, I guess these projects have quite some overlap with what testing-farm is doing for itself. But can ping testing-farm team to check these? At the point of when testing-farm is actually executing, the environment should be as vanilla as possible, and I think these changes should also be included in testing-farm by default.

Reply:
I have to agree with @LecrisUT: this requires an explanation. Comments must explain _why_, not what, and they belong in the code, not in a github discussion.